library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java

```diff
@@ -110,15 +108,14 @@ public void run() {
 
     private void handleNotificationBroadcast(Context context, Intent intent) {
         String action = intent.getAction();
+        long[] ids = intent.getLongArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS);
+        int[] statuses = intent.getIntArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES);
         switch (action) {
             case ACTION_LIST:
-                long[] ids = intent.getLongArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS);
-                int[] statuses = intent.getIntArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES);
                 sendNotificationClickedIntent(context, ids, statuses);
                 break;
             case ACTION_OPEN: {
```

Comment:
a completed download - we're not opening it but the action is to open it, we just delegate to the client instead of do it ourselves